homeassistant/setup.py

```diff
@@ -319,9 +317,15 @@ async def async_process_deps_reqs(
         raise HomeAssistantError("Could not set up all dependencies.")
 
     if not hass.config.skip_pip and integration.requirements:
-        await requirements.async_get_integration_with_requirements(
-            hass, integration.domain
-        )
+        if integration.domain in hass.timeout.zones:
```

Comment:
Feels a bit weird that we need to maintain this knowledge of how timeouts work outside of the timeout classes. I would just expect it to do nothing if we call `async_freeze` and there is no timer to freeze.

Reply:
I wanted it to throw an exception if we ask for to freeze a timeout with no timer as it was difficult to track down the problem when I was doing testing.   Maybe change this to `async_freeze_if_timer(integration.domain)` or `async_freeze(integration.domain, allow_notimer=True)` or `async_freeze_allow_notimer(integration.domain)` ?

Reply:
Edit: it may not actually matter anymore with the latest changes.  I'll retest and see if we can remove this if the wifi works inflight

Reply:
is fix anyway, I clean it up